src/box/applier.cc

```diff
@@ -2775,6 +2771,15 @@ applier_stop(struct applier *applier)
 	applier->fiber = NULL;
 }
 
+void
+applier_kill(struct applier *applier, struct error *e)
+{
+	assert(applier->fiber != NULL);
+	if (e != NULL)
+		applier_set_last_error(applier, e);
+	fiber_cancel(applier->fiber);
```

Comment:
Shouldn't the function nullify `applier->fiber`? So that it's impossible to call `applier_kill` several times in a row.